alpha/apps/kaltura/lib/KSecureEntryHelper.class.php

```diff
@@ -241,22 +230,36 @@ public function isAssetAllowed(asset $asset)
 	
 	public function shouldBlock()
 	{
-		return $this->hasBlockAction;
+		return $this->getActionList(RuleActionType::BLOCK);		
 	}
 	
 	protected function isFlavorParamsAllowed($flavorParamsId)
 	{
-		if($this->hasLimitFlavorsAction)
+		$actionList = $this->getActionList(RuleActionType::LIMIT_FLAVORS);
+		if ($actionList)
 		{
-			$flavorParamsIds = explode(',', $this->limitFlavorsAction->getFlavorParamsIds());
+			// take only the first LIMIT_FLAVORS action
+			$action = reset($actionList);
+			$flavorParamsIds = explode(',', $action->getFlavorParamsIds());
 			$exists = in_array($flavorParamsId, $flavorParamsIds);
-			if($this->limitFlavorsAction->getIsBlockedList())
+			if($action->getIsBlockedList())
 				return !$exists;
 			else 
 				return $exists;
 		}
 		return true;
 	}
+
+	public function updateDeliveryAttributes(DeliveryProfileDynamicAttributes $deliveryAttributes)
+	{
+		$actionList = $this->getActionList(RuleActionType::LIMIT_DELIVERY_PROFILES);
+		if ($actionList)
+		{
+			// take only the first LIMIT_DELIVERY_PROFILES action
+			$action = reset($actionList);
+			$deliveryAttributes->setDeliveryProfileIds($action->getDeliveryProfileIds(), $action->GetisBlockedList());
```

Comment:
Methods are not case sensitive, and yet, you probably meant to getIsBlockedList.